### DISCUSSION TITLE
[projects beta] - Migrate existing projects to beta version

### DISCUSSION THREAD

**Discussion by adriens** [community/community#6141] [Projects and Issues]
It would be very useful to be able to migrate exsting projects to the beta version, from a single click.

**Comment by galargh** [comment]
I created a tutorial on how to migrate from the legacy project boards to the new projects (beta): https://github.com/galargh/projects-migration#how-to-migrate-from-github-projects-to-github-projects-beta

TLDR: It describes how to migrate cards (including column names) by either running a GitHub Actions workflow or a bash script. Unfortunately, it does require some manual steps because projects (beta) API is not complete yet.

After the migration, I'm also heavily using this action I created https://github.com/protocol/github-api-action-library/tree/master/add-project-items-by-content-query. I use it to periodically populate my projects based on content search queries (e.g. I automated adding all the issues that mention me - https://github.com/galargh/.github/blob/684c316/.github/workflows/add-project-items.yml).

**Reply by adriens** [reply]
Hi @galargh , thanks for this very well documented anwser and linked resources 👍

**Reply by jamesxabregas** [reply]
Hi @galargh, I tried your migration scripts. Was usefull for bringing some of my issues across but I hit API limits so I ended up having to move some items across manually. 
Would be great if Github developed an official migration process.

**Comment by doidor** [comment]
I implemented python script that helps you migrate your project (similar to @galargh's solution) which uses the GraphQL api for the old projects too (might be more future proof) and also allows you to map column names and migrate PRs.

https://github.com/doidor/gh-projects-migration

**Comment by adriens** [comment]
See [today tweet](https://twitter.com/github/status/1539360979515019268) : 

![image](https://user-images.githubusercontent.com/5235127/174903079-ef061f6f-d30a-4dbb-8bc5-cb6da87cbcd7.png)

**Reply by kcollingridge** [reply]
I have several projects (classic) that I would like to migrate to the new projects style, which is no longer beta. How can I migrate them? The option to migrate does not seem to be available.

**Comment by fgnievinski** [comment]
URLs in notes are lost upon migration.
is this the right place to file this issue?

**Reply by fgnievinski** [reply]
actually, URLs are just hidden in the draft details.

**Comment by jge162** [comment]
you can reach out to GitHub support for assistance on migrating your project to the beta version. They may be able to provide guidance or help you troubleshoot any issues that arise during the migration process.

**Reply by kcollingridge** [reply]
The issue was not with migrating to beta (as it isn't beta anymore), but with migrating from classic to the new projects. I did eventually figure it out - you need to enable project migration in feature preview, then the option to migrate appears! Not sure why it is set up like this as it took a very long time to find why the option wasn't showing up, but I'm relieved not to have to do it manually.
https://docs.github.com/en/issues/planning-and-tracking-with-projects/creating-projects/migrating-from-projects-classic?apiVersion=2022-11-28

**Reply by jge162** [reply]
Nice.
